pkg/auth/agent_local_authenticator.go

```diff
@@ -86,6 +86,21 @@ func (a *AgentLocalAuthenticator) AuthAgentAuth(token string) (interface{}, erro
 	}
 	a.log.Infof("Authenticating infraEnv %s JWT", infraEnvID)
 
+	expiryDateTime, err := gencrypto.ParseExpirationFromToken(token)
+	// it is expected that agent installer sends the correct token
+	// and will contain the exp claim when it is required.
+	if err != nil && err.Error() != gencrypto.TokenMissingExpClaim {
```

Comment:
Additionally this error handling is flawed. If we know we don't have the exp claim, then it's safe to proceed, but if we have an error that isn't the missing claim we should error.